### DISCUSSION TITLE
Proposal for delivering docker image to docker hub

### DISCUSSION THREAD

**Discussion by hankquan** [Netflix/conductor#2629] [Ideas]
## bad experience
- It costs too much time to build docker image for conductor server and ui, and it failed many times either.

## proposal
Deliver conductor server and ui image with specific version(tag) to docker hub in every release, then users can easily prepare their development or even product environment instead of building themselves.

**Comment by hankquan** [comment]
If this proposal is agreed by owners, I will study how to implement it.

**Comment by jxu-nflx** [comment]
Hello @howaric thanks for the suggestion! That is exactly what we are planning to do, we will be publish images to [netflixoss](https://hub.docker.com/u/netflixoss)

**Reply by hankquan** [reply]
That's great, look forward to it.

**Comment by flavioschuindt** [comment]
Looking forward for it too. One question to consider: conductor is composed of different modules. This means for example I can have a module A (e.g. RedisLock) enabled for a user but in others it might not make sense. How are you guys considering it in those images? There is a tradeoff between extensability and image size here as this image can really grow the more modules are activated and built by Gradle.

**Reply by jxu-nflx** [reply]
Hello @flavioschuindt As a first step, we will be publishing the image with all modules built in, and users can config which modules to use on `CONFIG_PROP` file. In the long run, we are considering dynamically loaded lightweight images with configurable modules
